test/cljam/io/vcf/util/validator_test.clj

```diff
@@ -0,0 +1,132 @@
+          {:normal {:A [1 2]}} [:normal]
+          {:A {:type "Integer" :number 2}}))))
+
+  (testing "number check"
+    (is (not
+         (#'validator/check-each-samples
+          {:normal {:A [1 2]}} [:normal]
+          {:A {:type "Integer" :number 2}})))
+    (is (not
+         (#'validator/check-each-samples
+          {:normal {:A [1 2]}} [:normal]
+          {:A {:type "Integer" :number "A"}})))
+    (is (string?
+         (get-in
+          (#'validator/check-each-samples
+           {:normal {:A [1]}} [:normal]
+           {:A {:type "Integer" :number 2}})
+          [:normal :A]))))
+  (testing "not contain meta"
+    (#'validator/check-each-samples
+     {:normal {:A [1] :b [100]}} [:normal]
+     {:A {:type "Integer" :number 1}})))
+
+(deftest invalid-variant?-test
+  (testing "contig check"
+    (is (not
+         (validator/invalid-variant?
+          {:chr "chr10" :pos 10}
+          {:contig [{:id "chr10"}]}
+          base-header)))
+    (is (string?
+         (:chr
+          (validator/invalid-variant?
+           {:chr "c hr10" :pos 10}
+           {:contig [{:id "c hr10"}]}
+           base-header)))))
+  (testing "ref check"
+    (is (string?
+         (:ref
+          (validator/invalid-variant?
+           {:chr "chr10" :pos 10
+            :ref "AP"}
+           {:contig [{:id "chr10"}]}
+           base-header)))))
+  (testing "alt check"
+    (is (string?
+         (:alt
+          (validator/invalid-variant?
+           {:chr "chr10" :pos 10 :ref "A" :alt "AT"}
+           {:contig [{:id "chr10"}]}
+           base-header))))
+    (is (not
+         (validator/invalid-variant?
+          {:chr "chr10" :pos 10 :alt ["A" "<DEL>" nil]}
+          {:contig [{:id "chr10"}]}
+          base-header)))
+    (is (not
+         (validator/invalid-variant?
+          {:chr "chr10" :pos 10 :alt ["G]17:198982]"
+                                      "]13:123456]T"]}
+          {:contig [{:id "chr10"}]}
+          base-header))))
+  (testing "qual check"
+    (is (not
+         (validator/invalid-variant?
+          {:chr "chr10" :pos 10
+           :qual 0.1}
+          {:contig [{:id "chr10"}]}
+          base-header))))
+  (testing "filter check"
+    (is (not
+         (validator/invalid-variant?
+          {:chr "chr10" :pos 10
+           :filter :PASS}
+          {:contig [{:id "chr10"}]}
+          base-header)))
+    (is (not
+         (validator/invalid-variant?
+          {:chr "chr10" :pos 10
+           :filter [:q10 :s5]}
+          {:contig [{:id "chr10"}]}
+          base-header))))
+
+  (testing "info check"
+    (validator/invalid-variant?
+     {:chr "chr10" :pos 10
+      :info {:GT 100}}
+     {}
+     base-header))
+  (testing "format check"
+    (is (not
+         (validator/invalid-variant?
+          {:chr "chr10" :pos 10
+           :FORMAT ["A"]}
+          {:contig [{:id "chr10"}]}
+          base-header)))))
+
+(deftest make-validator-test
+  (is (thrown? clojure.lang.ExceptionInfo
```

Comment:
Check the expected ex-data explicitly. This helps users finding out the usage of the validator a lot.